ui/apps/platform/src/Containers/NetworkGraph/api/useModifyBaselineStatuses.ts

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react';
+import { markNetworkBaselineStatuses } from 'services/NetworkService';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+import { Flow } from '../types/flow.type';
+
+type FlowStatus = 'BASELINE' | 'ANOMALOUS';
+
+type Result = {
+    isModifying: boolean;
+    error: string;
+};
+
+type ModifyBaselineStatuses = {
+    modifyBaselineStatuses: (
+        flows: Flow[],
+        status: FlowStatus,
+        onSuccessCallback: () => void
+    ) => void;
+} & Result;
+
+const defaultResult = {
+    isModifying: false,
+    error: '',
+};
+
+function transformFlowsToPeers(flows: Flow[], status: FlowStatus) {
+    return flows.map((flow) => {
+        const { entityId, type, entity, namespace, direction, port, protocol } = flow;
+        const peer = {
+            peer: {
+                entity: {
+                    id: entityId,
+                    name: entity,
+                    namespace,
+                    type,
+                },
+                ingress: direction === 'Ingress',
+                port,
+                protocol,
+            },
+            status,
+        };
+        return peer;
+    });
+}
+
+function useModifyBaselineStatuses(deploymentId): ModifyBaselineStatuses {
+    const [result, setResult] = useState<Result>(defaultResult);
+
+    function modifyBaselineStatuses(
+        flows: Flow[],
+        status: FlowStatus,
+        onSuccessCallback: () => void
+    ) {
+        setResult({ isModifying: true, error: '' });
+        const peers = transformFlowsToPeers(flows, status);
+        const promise = markNetworkBaselineStatuses({
+            deploymentId,
+            networkBaselines: peers,
+        });
+        promise
```

Comment:
Same question as in other hook: why set the promise to a variable, just to then add the resolve handler?

Reply:
Yea that is a good point. We can remove it